### PR TITLE
Generic schema mapping

### PR DESCRIPTION
**Pull request by andyhite**
This PR makes the schema mapping process generic and included in the common lib.

### CODE REVIEW

File: extensions/purchase-label/functions/src/index.ts [hunk 1]
```diff
@@ -37,8 +39,13 @@ export const purchaseLabel = functions.handler.firestore.document.onWrite(
       handleUpdateDocument(change.after, update);
     } catch (err) {
       // Update the document with error information on failure
-      if ((err as Error).message)
-        handleUpdateDocument(change.after, { error: (err as Error).message });
+      if ((err as Error).message) {
+        handleUpdateDocument(change.after, {
+          [config.shippingLabelKey]: {
+            errors: (err as Error).message,
```
Comment (andyhite) [inline extensions/purchase-label/functions/src/index.ts:45]: This is unrelated to the purpose of this PR, but I realized I wasn't putting the errors in the correct key.

File: libs/common/src/utils/mapDataToSchema.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+export const mapDataToSchema = (data: DocumentData, schema: ParamSchema) => {
+  logs.mappingData(data, schema);
+
+  try {
+    return mapDataToSchema(data, schema);
+  } catch (err) {
+    logs.errorMappingData(err as Error);
+    throw err;
+  }
+};
```
Comment (andyhite) [inline libs/common/src/utils/mapDataToSchema.ts:38]: This is where the bulk of the work in this PR lives

File: libs/common/src/utils/mapDataToSchema.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { get, omit, castArray } from 'lodash';
+import { DocumentData } from '@google-cloud/firestore';
+import * as logs from '../logs';
+
+type ParamSchema = { [key: string]: any };
+
+const mapDataToSchemaWithoutLogging = (
```
Comment (forksofpower) [inline libs/common/src/utils/mapDataToSchema.ts:7]: Where is this being used?

**Comment by forksofpower** [comment]
lgtm :shipit:

File: libs/common/src/utils/mapDataToSchema.ts (hunk 3 above)
Comment (andyhite) [reply libs/common/src/utils/mapDataToSchema.ts:7]: It's used in the function below it that's actually exported. I didn't watch to have the try/catch at every recursive iteration of the function, so I wanted to wrap just the top-level of it in the try/catch.

**Pull request merged**